contracts/bulk-payments/src/bulk_payments.rs

```diff
@@ -0,0 +1,140 @@
+#![no_std]
+
+use multiversx_sc::derive_imports::*;
+use multiversx_sc::imports::*;
+
+const MAX_USERS_ALLOW: usize = 1000;
+const FOUR_HOURS: u64 = 60 * 60 * 4 * 1_000; // 4 hours
+
+#[type_abi]
+#[derive(NestedEncode, NestedDecode, TopEncode, TopDecode, Clone, PartialEq)]
+pub struct UserAddrTimestamp<M: ManagedTypeApi> {
+    pub addr: ManagedAddress<M>,
+    pub timestamp: u64,
+}
+
+#[multiversx_sc::contract]
+pub trait BulkPayments {
```

Comment:
Is BulkPayments a proper name for this contract. This makes me expect a different type of SC.

Reply:
Started as bulk payments, but now it does all the necessary actions but distributing. So I guess it's a storage for bulk payments.
We can add distribute also and keep the name.